### PR TITLE
[Mellanox] Adding SKU Mellanox-SN5600-C256

### PR DESCRIPTION
**Pull request by DavidZagury**
##### Work item tracking
- Microsoft ADO **(number only)**:

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by dprital** [comment]
@DavidZagury , Since CMIS Host Management should be enabled on this SKU, you need to add the following files as well:
* optict_si_settings.json
* media_settings.json

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256/pg_profile_lookup.ini [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  50000    5m 41984   19456  22528         0
+ 100000    5m 59392   19456  39936         0
+ 200000    5m 65536   19456  46080         0
+ 400000    5m 99328   19456  79872         0
+ 800000    5m 128000  38912  89088         0
+  10000   40m 36864   19456  17408         0
+  25000   40m 37888   19456  18432         0
+  40000   40m 41984   19456  22528         0
+  50000   40m 44032   19456  24576         0
+ 100000   40m 63488   19456  44032         0
```
Comment (stephenxs) [inline device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256/pg_profile_lookup.ini:31]: let's set data MTU as 4k.
Using the latest excel, I got xoff 49152

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256/pg_profile_lookup.ini [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ 200000   40m 74752   19456  55296         0
+ 400000   40m 117760  19456  98304         0
+ 800000   40m 163840  38912  124928        0
+  10000  300m 39936   19456  20480         0
+  25000  300m 47104   19456  27648         0
+  40000  300m 55296   19456  35840         0
+  50000  300m 61440   19456  41984         0
+ 100000  300m 97280   19456  77824         0
```
Comment (stephenxs) [inline device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256/pg_profile_lookup.ini:39]: I got xoff 83968

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256/pg_profile_lookup.ini (hunk 1 above)
Comment (DavidZagury) [reply device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256/pg_profile_lookup.ini]: @stephenxs fixed

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256/pg_profile_lookup.ini (hunk 2 above)
Comment (DavidZagury) [reply device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256/pg_profile_lookup.ini]: @stephenxs fixed

**Review by stephenxs — APPROVED** [review]

**Pull request closed without merging**